plugins/swiotrefactor/src/max14906/diodigitalchannelcontroller.cpp

```diff
@@ -0,0 +1,243 @@
+}
+
+void DioDigitalChannelController::createWriteCurrentLimitCommand(int index)
+{
+	QString text = m_digitalChannel->m_currentLimitsCombo->combo()->currentText();
+	Command *writeCurrentLimitCmd =
+		new IioChannelAttributeWrite(m_channel, "current_limit", text.toStdString().c_str(), nullptr);
+	connect(writeCurrentLimitCmd, &scopy::Command::finished, this,
+		&DioDigitalChannelController::writeCurrentLimitCmdFinished, Qt::QueuedConnection);
+	m_cmdQueue->enqueue(writeCurrentLimitCmd);
+}
+
+void DioDigitalChannelController::createWriteTypeCommand(int index)
+{
+	QString text = m_digitalChannel->m_configModesCombo->combo()->currentText();
+	m_type = text.toStdString();
+	Command *writeTypeCmd =
+		new IioChannelAttributeWrite(m_channel, m_iioAttrType.c_str(), text.toStdString().c_str(), nullptr);
+	connect(writeTypeCmd, &scopy::Command::finished, this, &DioDigitalChannelController::writeTypeCmdFinished,
+		Qt::QueuedConnection);
+	m_cmdQueue->enqueue(writeTypeCmd);
+}
+
+void DioDigitalChannelController::readAvailableTypeCmdFinished(Command *cmd)
+{
+	IioChannelAttributeRead *tcmd = dynamic_cast<IioChannelAttributeRead *>(cmd);
+	if(!tcmd) {
+		return;
+	}
+	if(tcmd->getReturnCode() >= 0) {
+		char *result = tcmd->getResult();
+		std::stringstream ss(result);
+		std::string aux;
+		while(std::getline(ss, aux, ' ')) {
+			m_availableTypes.push_back(aux);
+		}
+		m_digitalChannel->setConfigModes(m_availableTypes);
+	} else {
+		qCritical(CAT_SWIOT_MAX14906)
+			<< "Could not read the available types, error code: " << tcmd->getReturnCode();
+	}
+}
+
+void DioDigitalChannelController::readTypeCmdFinished(Command *cmd)
+{
+	IioChannelAttributeRead *tcmd = dynamic_cast<IioChannelAttributeRead *>(cmd);
+	if(!tcmd) {
+		return;
+	}
+	if(tcmd->getReturnCode() >= 0) {
+		char *result = tcmd->getResult();
+		m_type = std::string(result);
+		m_digitalChannel->setSelectedConfigMode(QString::fromStdString(m_type));
+	} else {
+		qCritical(CAT_SWIOT_MAX14906) << "Could not read the type, error code: " << tcmd->getReturnCode();
+	}
+}
+
+void DioDigitalChannelController::readRawCmdFinished(Command *cmd)
+{
+	IioChannelAttributeRead *tcmd = dynamic_cast<IioChannelAttributeRead *>(cmd);
+	if(!tcmd) {
+		return;
+	}
+	if(tcmd->getReturnCode() >= 0) {
+		char *result = tcmd->getResult();
+		bool ok = false;
+		bool rawValue = QString(result).toInt(&ok);
+		if(ok) {
+			m_digitalChannel->m_valueSwitch->setChecked(rawValue);
+		}
+	} else {
+		qCritical(CAT_SWIOT_MAX14906)
+			<< "Could not read initial channel raw value, error code: " << tcmd->getReturnCode();
+	}
+}
+
+void DioDigitalChannelController::readCurrentLimitCmdFinished(Command *cmd)
+{
+	IioChannelAttributeRead *tcmd = dynamic_cast<IioChannelAttributeRead *>(cmd);
+	if(!tcmd) {
+		return;
+	}
+	if(tcmd->getReturnCode() >= 0) {
+		char *result = tcmd->getResult();
+		m_digitalChannel->m_currentLimitsCombo->combo()->setCurrentIndex(
+			m_digitalChannel->m_currentLimitsCombo->combo()->findText(result));
```

Comment:
Should we validate the result of findText? If the result from IIO is an unexpected value, the find function will return -1 and the current index settings might have weird behaviour.